src/darker/__main__.py

```diff
@@ -56,45 +60,70 @@ def format_edited_parts(
     :param enable_isort: ``True`` to also run ``isort`` first on each changed file
     :param black_config: Configuration to use for running Black
     :param report_unmodified: ``True`` to yield also files which weren't modified
+    :param jobs: number of cpu processes to use (0 - autodetect)
     :return: A generator which yields details about changes for each file which should
              be reformatted, and skips unchanged files.
 
     """
     files_to_blacken = apply_black_excludes(changed_files, git_root, black_config)
-    for path_in_repo in sorted(changed_files):
-        src = git_root / path_in_repo
-        rev2_content = git_get_content_at_revision(
-            path_in_repo, revrange.rev2, git_root
-        )
-
-        # 1. run isort
-        if enable_isort:
-            rev2_isorted = apply_isort(
-                rev2_content,
-                src,
-                black_config.get("config"),
-                black_config.get("line_length"),
-            )
-        else:
-            rev2_isorted = rev2_content
-        if src in files_to_blacken:
-            # 9. A re-formatted Python file which produces an identical AST was
-            #    created successfully - write an updated file or print the diff if
-            #    there were any changes to the original
-            content_after_reformatting = _reformat_single_file(
+    with concurrent.futures.ThreadPoolExecutor(max_workers=jobs or None) as executor:
+        futures: List[
+            concurrent.futures.Future[ProcessedDocument]
+        ] = []  # pylint: disable=unsubscriptable-object
+        for path_in_repo in sorted(changed_files):
+            enable_black = (git_root / path_in_repo) in files_to_blacken
+            future = executor.submit(
+                _format_single_file,
                 git_root,
                 path_in_repo,
                 revrange,
-                rev2_content,
-                rev2_isorted,
+                black_config if enable_black else None,
                 enable_isort,
-                black_config,
             )
-        else:
-            # File was excluded by Black configuration, don't reformat
-            content_after_reformatting = rev2_isorted
-        if report_unmodified or content_after_reformatting != rev2_content:
-            yield (src, rev2_content, content_after_reformatting)
+            futures.append(future)
+
+        for future in concurrent.futures.as_completed(futures):
+            src, rev2_content, content_after_reformatting = future.result()
+            if report_unmodified or content_after_reformatting != rev2_content:
+                yield (src, rev2_content, content_after_reformatting)
+
+
+def _format_single_file(
+    git_root: Path,
+    path_in_repo: Path,
+    revrange: RevisionRange,
+    black_config: BlackConfig,
```

Comment:
Since `format_edited_parts` can pass `black_config==None`, I believe this should be: 
```suggestion
    black_config: Optional[BlackConfig],
```

Reply:
See below, probably not after all. Instead, add a separate `enable_black: bool` argument.